tests/tests_unit/test_base.py

```diff
@@ -138,6 +140,11 @@ def use(self):
         return self._cognite_client
 
 
+@pytest.fixture(scope="session")
+def cognite_mock_client() -> CogniteClientMock:
+    return CogniteClientMock()
```

Comment:
The mock is very expensive to instantiate, which is why I did it. How can this lead to errors?

Reply:
Because the mock is stateful:
```py
@pytest.fixture(scope="session")
def cognite_mock_client() -> CogniteClientMock:
    return CogniteClientMock()


def test_xyz_1(cognite_mock_client):
    from cognite.client.data_classes import Asset

    azzet = Asset(parent_id=123, cognite_client=cognite_mock_client)
    _ = azzet.parent()
    cognite_mock_client.assets.retrieve.assert_called_once_with(id=123)


def test_xyz_2(cognite_mock_client):
    cognite_mock_client.assets.retrieve(external_id="foo")
    cognite_mock_client.assets.retrieve.assert_called_once_with(external_id="foo")
    # AssertionError: Expected 'retrieve' to be called once. Called 2 times.
    # Calls: [call(id=123), call(external_id='foo')]
```